airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/formats/jsonl_parser.py

```diff
@@ -0,0 +1,72 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, BinaryIO, Iterator, Mapping, TextIO, Union
+
+import pyarrow as pa
+from pyarrow import json as pa_json
+
+from .abstract_file_parser import AbstractFileParser
+from .json_spec import JsonlFormat
+
+
+class JsonlParser(AbstractFileParser):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.format_model = None
+        self._inferred_schema = None
+
+    @property
+    def is_binary(self) -> bool:
+        return True
+
+    @property
+    def format(self) -> JsonlFormat:
+        if self.format_model is None:
+            self.format_model = JsonlFormat.parse_obj(self._format)
+        return self.format_model
+
+    def _read_options(self) -> Mapping[str, str]:
+        """
+        https://arrow.apache.org/docs/python/generated/pyarrow.json.ReadOptions.html
+        build ReadOptions object like: pa.json.ReadOptions(**self._read_options())
+        """
+        return {**{"block_size": self.format.block_size, "use_threads": True}}
+
+    def _parse_options(self) -> Mapping[str, str]:
+        """
+        https://arrow.apache.org/docs/python/generated/pyarrow.json.ParseOptions.html
+         build ParseOptions object like: pa.json.ParseOptions(**self._parse_options())
+        """
+        parse_options = {
+            "newlines_in_values": self.format.newlines_in_values,
+            "unexpected_field_behavior": self.format.unexpected_field_behavior,
+        }
+        if self._inferred_schema:
+            parse_options["explicit_schema"] = self._inferred_schema
```

Comment:
That also means removing this

Reply:
I have removed above logic but, need to pass explicit schema to json parser.  let me know if I am missing something.